src/Common/Core/Impl/Services/ServiceBag.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Common.Core.Services {
+    public sealed class ServiceBag: IServiceBag {
```

Comment:
It's a good question what should be the main entry point. `ICoreShell` is already imported through MEF. If we are adding `ICoreServices`, I'd rather make explicit properties for all the required types (`FileSystem`, `Registry`, `Log`, `Telemetry`) and add a `ICoreShell.CoreServices` property. 
`ICoreShell` itself requires a split - it contains methods that are related to the UI thread that should not be available in Common and RHost.Client, but that is a separate task.

---
In reply to: [80990180](https://github.com/Microsoft/RTVS/pull/2408#discussion_r80990180) [](ancestors = 80990180)